lib/Transforms/MemCap/FoldCapIntrinsics.cpp

```diff
@@ -37,7 +38,72 @@ class MemCapFoldIntrinsics : public ModulePass {
   Function *IncOffset;
   Function *SetOffset;
   Function *GetOffset;
+  Function *GetBase;
+  Function *GetLength;
+  Function *GetType;
+  Function *GetPerms;
+  Function *GetTag;
+  Function *GetSealed;
+
   bool Modified;
+
+  template <typename Infer> void foldGet(Function *Intrinsic, Infer infer) {
```

Comment:
The build system currently passes -std=c++11, so it gives me `error: 'auto' not allowed in lambda parameter`